crates/types/src/blockchain/header/v1.rs

```diff
@@ -29,7 +31,7 @@ pub struct BlockHeaderV1 {
     /// The header metadata calculated during creation.
     /// The field is pub(crate) to enforce the use of the [`PartialBlockHeader::generate`] method.
     #[cfg_attr(feature = "serde", serde(skip))]
-    #[derivative(PartialEq = "ignore")]
+    #[educe(PartialEq(ignore))]
```

Comment:
Not related to this PR, just a side question because I was wondering if it could be better to just write `PartialEq` manually instead of generating it with a macro from a separate crate? Basically this boils down to the rule: use classic `derive`, unless you can't, then write the implementation manually.

I'd be interested to hear opinions.

Reply:
Mhh interesting question.
Personally I perfer to use the crate because the ignore is directly on the structure and so you can read everything here compare to when you have to write the implementation yourself it's separated code.

Also with the crate we ensure that all fields that are not explicitly ignored are used which is up to the developers and reviewers to ensure that when writing by hand which lead to more errors IMO.